File: PRACTICA/PRACTICA FINAL/QytetetRuby/lib/titulo_propiedad.rb
#encoding: utf-8
# der, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module ModeloQytetet
  
  # Requieres/ incluides
  require_relative "casilla"
  require_relative "titulo_propiedad"
  
  class TituloPropiedad
    
    # Getter/consultores 
    attr_reader :nombre, :alquiler_base, :factor_revalorizado, :hipoteca_base, :precio_edificar, :casilla, :propietario
    
    # Getter and Setter
    attr_accessor :hipotecada
    
    # Setter/Modificadores
    attr_writer :casilla, :propietario
    
    # Inicialize
      def initialize(n, h, ab, fr, hp, pe, ca=nil, pro=nil)
        @nombre = n
        @hipotecada = h
        @alquiler_base =ab
        @factor_revalorizado = fr
        @hipoteca_base= hp
        @precio_edificar =pe
        
        @casilla= ca
        @propietario= pro
        

        
      end
      
    # Metodos propios de la clase
      def cobrar_alquiler(coste)
        
        raise"......."
        
      end
      
      def propietario_encarcelado?
        
        raise"......"
               
      end
      
      def tengo_propietario?
        
        raise "....."
        
      end
    #Metodo to string
      def to_s
        "Texto: #{@nombre} \n Hipotecada: #{@hipotecada} \n Alquiler base: #{@alquiler_base} \n
        Factor Revalorizado: #{@factor_revalorizade} \n
        Hipoteca base: #{@hipoteca_base} \n
        Precio a edificar: #{@precio_edifica} \n "
      end 
  end
end

File: PRACTICA/PRACTICA FINAL/QytetetJava/src/modeloqytetet/TituloPropiedad.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
/*   */
package modeloqytetet;

/**
 *
 * @author franmerchan6
 */
import modeloqytetet.TituloPropiedad;
import modeloqytetet.Casilla;

public class TituloPropiedad {
    
    /* Variables  */
    private String nombre;
    private boolean hipotecada;
    private int alquilerBase;
    private double factorRevaloracion;
    private int hipotecaBase;
    private int precioEdificar;
    private Jugador propietario = null;
    private Casilla casilla;

    /* Getter/consultores  */
    
    String getNombre() {
        return nombre;
    }

    int getAlquilerBase() {
        return alquilerBase;
    }

    double getFactorRevaloracion() {
        return factorRevaloracion;
    }

    int getHipotecaBase() {
        return hipotecaBase;
    }

    int getPrecioEdificar() {
        return precioEdificar;
    }
    
    boolean getHipotecada(){
        return hipotecada;
    }

    public Jugador getPropietario() {
        return propietario;
    }

    public Casilla getCasilla() {
        return casilla;
    }
    
    

    /* Setter / Modificadores  */
    
    public void setHipotecada(boolean hipotecada) {
        this.hipotecada = hipotecada;
    }
    
    void setPropietario(Jugador propietario){
         this.propietario =  propietario;
    }
    
    void setCasilla(Casilla casilla){  
        this.casilla = casilla;
    }

    /* Contructores  */
    public TituloPropiedad(String nombre, boolean hipotecada, int alquilerBase, double factorRevaloracion, int hipotecaBase, int precioEdificar) {
        this.nombre = nombre;
        this.hipotecada = hipotecada;
        this.alquilerBase = alquilerBase;
        this.factorRevaloracion = factorRevaloracion;
        this.hipotecaBase = hipotecaBase;
        this.precioEdificar = precioEdificar;
        this.propietario =null;
        
    }
    
    /* Funciones expecificas de la clase  */
    
   //modicifacion 08-11-2017
    public void cobrarAlquiler(int coste){
        
        this.getPropietario().modificarSaldo(coste);
        
    }
    //FIN modicifacion 08-11-2017
    
    boolean propietarioEncarcelado(){

       throw new UnsupportedOperationException("Sin implementar"); 
        
    }
    
    
    boolean tengoPropietario (){
        
        throw new UnsupportedOperationException("Sin implementar");
        
    }
    
    /* Funcion to string  */
    @Override
    public String toString() {
        return "TituloPropiedad{" + "nombre=" + nombre + ", hipotecada= " + hipotecada + ", alquilerBase= " + alquilerBase + ", factorRevaloracion= " + factorRevaloracion + ", hipotecaBase= " + hipotecaBase + ", precioEdificar= " + precioEdificar + ", Casilla= "+ casilla + '}';
        

}
    
    
    
}
